commit 4e319e970a76a195987ab1b7e988bb0d58e3a6d6
Author: Eric Anholt <eric@anholt.net>
Date:   Mon Mar 6 12:17:16 2017 -0800

    panel-raspberrypi-touchscreen: Round up clk rate to fix DSI panel.
    
    Commit 488f9bc8e3def93e0baef53cee2026c2cb0d8956 slightly increased the
    reported rate of PLLD, so the clk driver decided that PLLD/3/8 was now
    higher than our requested pixel clock rate and rejected it in favor of
    PLLD/4/8, which then ran the pixel clock way out of spec.
    
    By bumping the requested clock rate just slightly, we get back to
    PLLD/3/8 like we wanted and the panel displays content again.
    
    Signed-off-by: Eric Anholt <eric@anholt.net>

diff --git a/drivers/gpu/drm/panel/panel-raspberrypi-touchscreen.c b/drivers/gpu/drm/panel/panel-raspberrypi-touchscreen.c
index 5d6bfd81b07a..b57b7ad9aa4c 100644
--- a/drivers/gpu/drm/panel/panel-raspberrypi-touchscreen.c
+++ b/drivers/gpu/drm/panel/panel-raspberrypi-touchscreen.c
@@ -220,7 +220,12 @@ static const struct drm_display_mode rpi_touchscreen_modes[] = {
 #define HBP         46
 #define HFP         ((PIXEL_CLOCK / (VTOTAL * VREFRESH)) - (HACT + HSW + HBP))
 
-		.clock = PIXEL_CLOCK / 1000,
+		/* Round up the pixel clock a bit (10khz), so that the
+		 * "don't run things faster than the requested clock
+		 * rate" rule of the clk driver doesn't reject the
+		 * divide-by-3 mode due to rounding error.
+		 */
+		.clock = PIXEL_CLOCK / 1000 + 10,
 		.hdisplay = HACT,
 		.hsync_start = HACT + HFP,
 		.hsync_end = HACT + HFP + HSW,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1137_4e319e970a76a195987ab1b7e988bb0d58e3a6d6.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1137_4e319e970a76a195987ab1b7e988bb0d58e3a6d6.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
